Upstream-Status:Inappropriate [debian patch]

From: Niko Tyni <ntyni@debian.org>
Subject: Work around an ICE on ia64
Closes: 548943

Temporarily work around an internal compiler error in Devel::PPPort
on ia64+gcc-4.3.


---
 cpan/Devel-PPPort/Makefile.PL |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/cpan/Devel-PPPort/Makefile.PL b/cpan/Devel-PPPort/Makefile.PL
index 67eebc1..f1ef7a2 100644
--- a/cpan/Devel-PPPort/Makefile.PL
+++ b/cpan/Devel-PPPort/Makefile.PL
@@ -75,6 +75,13 @@ sub configure
     push @moreopts, INSTALLDIRS => ($] >= 5.007003 ? 'perl' : 'site');
   }
 
+
+  # temporary Debian hack, see http://bugs.debian.org/548943
+  require Config;
+  if ($Config::Config{archname} =~ /^ia64/) {
+    push @moreopts, OPTIMIZE => '-g -O0';
+  }
+
   if ($opt{'apicheck'}) {
     $PL_FILES{'apicheck_c.PL'} = 'apicheck.c';
     push @C_FILES, qw{ apicheck.c };
-- 
tg: (a508b62..) debian/devel-ppport-ia64-optim (depends on: upstream)
